packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -45,31 +45,15 @@ export interface QueryParams {
     shortCode?: string
 }
 
-export type QueryResponse<T> =
-    | {
-          isLoading: boolean
-          error?: undefined
-          data?: undefined
-      }
-    | {
-          isLoading: true
-          error?: Error
-          data?: T
-      }
-    | {
-          isLoading: false
-          error: Error
-          data?: undefined
-      }
-    | {
-          isLoading: false
-          error?: undefined
-          data: T
-      }
+export type QueryResponse<T> = {
+    isLoading: boolean
+    error?: Error | undefined
+    data?: T | undefined
+}
```

Comment:
typescript complains the union type, I'm not sure how to deal with it instead of having a generic one.